jujucrashdump/crashdump.py

```diff
@@ -52,10 +53,13 @@
 
 
 def retrieve_single_unit_tarball(tuple_input):
-    unique, machine, alias_group = tuple_input
+    unique, machine, alias_group, all_machines = tuple_input
     unit_unique = uuid.uuid4()
-    juju_cmd("scp %s:/tmp/juju-dump-%s.tar %s.tar"
-             % (machine, unique, unit_unique))
+    scp_cmd = 'scp -o StrictHostKeyChecking=no'
+    for ip in all_machines[machine]:
```

Comment:
I'm a bit confused by this loop.  The function purports to retrieve the tarball from a single unit, but here you're looping over every single machine.  Additionally, this function is already called for every machine (across multiple processes) by `retrieve_unit_tarballs`, so this seems like it's just duplicating work.  Am I missing something here?

Reply:
all_machines contains a dictionary with {machineid: [all known IPs of that machine]}. This loop goes through all known IPs of a given machine (all_machines[machine]) and when it finds one which works, it will break the loop.